src/Dscp/DB/SQLite/Queries.hs

```diff
@@ -0,0 +1,313 @@
+assertJust action message = do
+    mb <- action
+
+    whenNothing mb $ do
+        throwIO message
+
+existsCourse :: DBM m => CourseId -> m Bool
+existsCourse course = do
+    exists existsCourseQuery (Only course)
+  where
+    existsCourseQuery = [q|
+        select  1
+        from    Courses
+        where   id = ?
+    |]
+
+existsStudent :: DBM m => StudentId -> m Bool
+existsStudent student = do
+    exists existsCourseQuery (Only student)
+  where
+    existsCourseQuery = [q|
+        select  1
+        from    Students
+        where   addr = ?
+    |]
+
+createStudent :: DBM m => StudentId -> m StudentId
+createStudent student = do
+    execute createStudentRequest (Only student)
+    return student
+  where
+    createStudentRequest = [q|
+        insert into  Students
+        values       (?)
+    |]
+
+createAssignment :: DBM m => Assignment -> m (Hash Assignment)
+createAssignment assignment = do
+    let courseId = assignment^.aCourseId
+
+    courseExists <- existsCourse courseId
+
+    unless courseExists $ do
+        throwM (CourseDoesNotExist courseId)
+
+    execute createAssignmentRequest
+        ( assignmentHash
+        , assignment^.aCourseId
+        , assignment^.aType
+        , assignment^.aAssignment
+        )
+    return assignmentHash
+  where
+    createAssignmentRequest = [q|
+        insert into  Assignments
+        values       (?,?,?,?)
+    |]
+    assignmentHash = hash assignment
+
+getAssignment :: DBM m => Hash Assignment -> m (Maybe Assignment)
+getAssignment assignmentHash = do
+    listToMaybe <$> query getAssignmentQuery (Only assignmentHash)
+  where
+    getAssignmentQuery = [q|
+        select  course_id, contents_hash, desc
+        from    Assignments
+        where   hash = ?
+    |]
+
+getSignedSubmission :: DBM m => PublicKey -> Hash Submission -> m (Maybe SignedSubmission)
+getSignedSubmission pk submissionHash = do
+    (listToMaybe . map ($ pk)) <$> query getSignedSubmissionQuery (Only submissionHash)
+  where
+    getSignedSubmissionQuery = [q|
+        select     student_addr,
+                   Submissions.contents_hash,
+                   Assignments.course_id,
+                   Assignments.contents_hash,
+                   Assignments.desc,
+                   Submissions.signature
+
+        from       Submissions
+
+        left join  Assignments
+               on  Assignments.hash = assignment_hash
+
+        where      Submissions.hash = ?
+    |]
+
+createTransaction :: DBM m => PublicKey -> PrivateTx -> m (PrivateTxId)
+createTransaction pk trans = do
+    transaction $ do
+        let ptid    = hash trans
+            subHash = trans^.ptSignedSubmission.to hash
+
+        _ <- getSignedSubmission pk ()
+
+        execute createTransactionRequest
+            ( ptid
+            , subHash
+            , trans^.ptGrade
+            , trans^.ptTime
+            , -1
+            )
+  where
+    createTransactionRequest :: Query
+    createTransactionRequest = [q|
+
```

Comment:
Seems like WIP, right? Mark it with `TODO` comment, if you're not going to implement this in scope of this PR, and create a corresponding issue.